spyderlib/plugins/shortcuts.py

```diff
@@ -487,20 +470,24 @@ def data(self, index, role=Qt.DisplayRole):
             elif column == NAME:
                 color = self.text_color
                 if self._parent == QApplication.focusWidget():
-                    if self.current_index().row() == index.row():
+                    if self.current_index().row() == row:
                         color = self.text_color_highlight
                     else:
                         color = self.text_color
-                return to_qvariant(self._enrich_text(shortcut.name, color))
+                text = self.rich_text[row]
+                text = '<p style="color:{0}">{1}</p>'.format(color, text)
+                return to_qvariant(text)
             elif column == SEQUENCE:
                 text = QKeySequence(key).toString(QKeySequence.NativeText)
                 return to_qvariant(text)
+            elif column == SEARCH_SCORE:

```

Comment:
Well call it what you want ;-).... but the idea is that to filter what is displayed in the `QTableView` I use a `SortFilterProxy` subclass which filters (based on a regex in this case), this handles the rows that get displayed... now for the order...

...having a column that stores the score makes the code simpler cause I can rely on the builtin sort that comes by default in the `SortFilterProxy`.